cmd/storage_bucket_replication_put.go

```diff
@@ -0,0 +1,85 @@
+package cmd
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/exoscale/cli/pkg/storage/sos"
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	storageBucketReplicationPutCmd.Flags().StringP(zoneFlagLong, zoneFlagShort, "", zoneFlagMsg)
+	storageBucketReplicationCmd.AddCommand(storageBucketReplicationPutCmd)
+}
+
+var storageBucketReplicationPutCmd = &cobra.Command{
+	Use:   "put sos://BUCKET file://replication.json",
+	Short: "Put replication configuration",
+	Long: `Put a replication configuration for a bucket. Bucket versioning needs to be enabled
+for both source and target bucket
+
+Example of a valid replication configuration:
+{
+    "Role": "role-uuid",
+    "Rules": [
+        {
+            "Status": "Enabled",
+            "Priority": 1,
+            "DeleteMarkerReplication": { "Status": "Disabled" },
+            "Filter" : { "Prefix": ""},
+            "Destination": {
+                "Bucket": "target-bucket"
+            },
+            "ID": "foo"
+        }
+    ]
+}
+
+More information at https://docs.aws.amazon.com/cli/latest/reference/s3api/put-bucket-replication.html#options & https://docs.aws.amazon.com/AmazonS3/latest/API/API_PutBucketReplication.html`,
+	Args: cobra.ExactArgs(2),
+	PreRunE: func(cmd *cobra.Command, args []string) error {
+
+		args[0] = strings.TrimPrefix(args[0], sos.BucketPrefix)
+		args[1] = strings.TrimPrefix(args[1], "file://")
```

Comment:
Specifying a `file://` protocol is not a pattern we use anywhere in the CLI codebase.

We should aim for a consistency here. If we take `exo c ssh-key register` as example you can just document second argument as file path (without protocol) and call it a day.

Alternative approach (for JSON payload) is one used by `exo iam org-policy update`, where payload can be specified as argument directly  "quoted JSON string" or piped to SDTIN if `-` is specified.

Reply:
a96caee9